godtools/App/Features/AppLanguage/Presentation/AppLanguages/AppLanguagesViewModel.swift

```diff
@@ -12,21 +12,58 @@ import Combine
 class AppLanguagesViewModel: ObservableObject {
     
     private let getAppLanguagesListUseCase: GetAppLanguagesListUseCase
+    private let getInterfaceStringInAppLanguageUseCase: GetInterfaceStringInAppLanguageUseCase
+    private let searchAppLanguageInAppLanguagesListUseCase: SearchAppLanguageInAppLanguagesListUseCase
     
     private var cancellables: Set<AnyCancellable> = Set()
     
     private weak var flowDelegate: FlowDelegate?
+    private let searchTextPublisher: CurrentValueSubject<String, Never> = CurrentValueSubject("")
+    private var allAppLanguages: [AppLanguageListItemDomainModel] = Array()
     
-    @Published var appLanguages: [AppLanguageListItemDomainModel] = Array()
+    @Published var appLanguageSearchResults: [AppLanguageListItemDomainModel] = Array()
+    @Published var navTitle: String = ""
     
-    init(flowDelegate: FlowDelegate, getAppLanguagesListUseCase: GetAppLanguagesListUseCase) {
+    init(flowDelegate: FlowDelegate, getAppLanguagesListUseCase: GetAppLanguagesListUseCase, getInterfaceStringInAppLanguageUseCase: GetInterfaceStringInAppLanguageUseCase, searchAppLanguageInAppLanguagesListUseCase: SearchAppLanguageInAppLanguagesListUseCase) {
         
         self.flowDelegate = flowDelegate
         self.getAppLanguagesListUseCase = getAppLanguagesListUseCase
+        self.getInterfaceStringInAppLanguageUseCase = getInterfaceStringInAppLanguageUseCase
+        self.searchAppLanguageInAppLanguagesListUseCase = searchAppLanguageInAppLanguagesListUseCase
+                
+        getInterfaceStringInAppLanguageUseCase
+            .observeStringChangedPublisher(id: AppLanguageStringKeys.AppLanguages.navTitle.rawValue)
+            .receive(on: DispatchQueue.main)
+            .assign(to: &$navTitle)
         
         getAppLanguagesListUseCase.observeAppLanguagesListPublisher()
             .receive(on: DispatchQueue.main)
-            .assign(to: &$appLanguages)
+            .sink { [weak self] (appLanguagesList: [AppLanguageListItemDomainModel]) in
+                
+                self?.allAppLanguages = appLanguagesList
+                self?.setSearchResults()
+            }
+            .store(in: &cancellables)
+        
+        searchTextPublisher
```

Comment:
I need to look at this a little more, but I'm thinking I'd like the search string to come into the ViewModel as an input.  That would cause some changes to the SearchBarView and SearchBarViewModel though.  

The idea is that the SearchBarView would contain a closure for the search string and then the AppLanguagesView would push that string to the viewModel as an input.  Going to look at this a little more first and see how it's working with the ToolFiltering because I know it would have to change there as well which could be an issue. But part of the View responsibility is pushing user input to the ViewModel.